File: app/src/main/java/com/haohao/kotlintest/help/ExtraDataHelper.java
package com.haohao.kotlintest.help;

import android.os.Bundle;

import com.haohao.kotlintest.fragment.NewsListFragment;
import com.haohao.kotlintest.fragment.ViewPagerFragment;
import com.haohao.kotlintest.util.HeadlineType;

import androidx.fragment.app.Fragment;

import static com.haohao.kotlintest.CommonConstant.CATEGORY_BBC_VIDEO;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_BRITISH_ENGLISH;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_ENGLISH_HEADLINE;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_HOW_TO_SAY;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_TED_SPEECH;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_TOPIC_VIDEO;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_VOA_SPECIAL;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_VOA_STANDARD;
import static com.haohao.kotlintest.CommonConstant.CATEGORY_VOA_VIDEO;


public class ExtraDataHelper {
    public static Fragment buildExtraFragment(String[] typeNamns, int[] typeCodes) {
        String type = "";
        switch (CATEGORY_VOA_SPECIAL){//InfoHelper.getInstance().getCategory()
            case CATEGORY_VOA_SPECIAL:
                type =HeadlineType.VOA;
                break;
            case CATEGORY_VOA_STANDARD:
                type =HeadlineType.CSVOA;
                break;
            case CATEGORY_VOA_VIDEO:
                type =HeadlineType.VOAVIDEO;
                break;
            case CATEGORY_HOW_TO_SAY:
                type =HeadlineType.MEIYU;
                break;
            case CATEGORY_BRITISH_ENGLISH:
                type =HeadlineType.BBC;
                break;
            case CATEGORY_ENGLISH_HEADLINE:
                type =HeadlineType.NEWS;
                break;
            case CATEGORY_TED_SPEECH:
                type =HeadlineType.TED;
                break;
            case CATEGORY_TOPIC_VIDEO:
                type =HeadlineType.TOPVIDEOS;
                break;
            case CATEGORY_BBC_VIDEO:
                type =HeadlineType.BBCWORDVIDEO;
                break;
                default:break;
        }
        String[] types1 = new String[]{
                type
        };
        Bundle bundle = ViewPagerFragment.Companion.buildArguments(15, HolderType.SMALL, types1,typeNamns,typeCodes);
        return ViewPagerFragment.Companion.newInstance(bundle);
    }
}

File: app/src/main/java/com/haohao/kotlintest/help/ExtraDataHelper.kt
package com.haohao.kotlintest.help

import androidx.fragment.app.Fragment
import com.haohao.kotlintest.CommonConstant.*
import com.haohao.kotlintest.fragment.ViewPagerFragment
import com.haohao.kotlintest.util.HeadlineType


object ExtraDataHelper {
    fun buildExtraFragment(typeNames: Array<String>, typeCodes: IntArray): Fragment {
        var type = ""
        when (InfoHelper.getInstance().getCategory()) {
            CATEGORY_VOA_SPECIAL -> type = HeadlineType.VOA
            CATEGORY_VOA_STANDARD -> type = HeadlineType.CSVOA
            CATEGORY_VOA_VIDEO -> type = HeadlineType.VOAVIDEO
            CATEGORY_HOW_TO_SAY -> type = HeadlineType.MEIYU
            CATEGORY_BRITISH_ENGLISH -> type = HeadlineType.BBC
            CATEGORY_ENGLISH_HEADLINE -> type = HeadlineType.NEWS
            CATEGORY_TED_SPEECH -> type = HeadlineType.TED
            CATEGORY_TOPIC_VIDEO -> type = HeadlineType.TOPVIDEOS
            CATEGORY_BBC_VIDEO -> type = HeadlineType.BBCWORDVIDEO
            else -> {
            }
        }
        val types1 = arrayOf(type)
        val bundle = ViewPagerFragment.buildArguments(15, HolderType.SMALL, types1, typeNames, typeCodes)
        return ViewPagerFragment.newInstance(bundle)
    }
}
